splitio/storage/__init__.py

```diff
@@ -354,4 +354,75 @@ def intersect(self, flag_sets):
         if not isinstance(flag_sets, set) or len(flag_sets) == 0:
             return False
 
-        return any(self.flag_sets.intersection(flag_sets))
\ No newline at end of file
+        return any(self.flag_sets.intersection(flag_sets))
+    
+class RuleBasedSegmentsStorage(object, metaclass=abc.ABCMeta):
+    """SplitRule based segment storage interface implemented as an abstract class."""
+ 
+    @abc.abstractmethod
+    def get(self, segment_name):
+        """
+        Retrieve a rule based segment.
+
+        :param segment_name: Name of the segment to fetch.
+        :type segment_name: str
+
+        :rtype: str
+        """
+        pass
+
+    @abc.abstractmethod
+    def update(self, to_add, to_delete, new_change_number):
```

Comment:
Probably not required in server-side, but the `update` method might return a boolean indicating whether the storage was actually modified or not. In client-side SDKs, this boolean is used as condition to emit the SDK_UPDATE event.